purchase_stock_picking_show_currency_rate/models/stock_picking.py

```diff
@@ -20,6 +20,12 @@ class StockPicking(models.Model):
         compute="_compute_currency_rate_amount",
         digits=0,
     )
+    rate_date_at_done = fields.Date(
+        help="Technical Field to store the rate date at done."
+    )
```

Comment:
@AungKoKoLin1997 What's the advantage of keeping the date instead of the actual rate? I've been suggesting that we should keep the actual rate (as we don't want to rely on the rate master for done transactions). Please provide your reasons if you think otherwise.